docs/SelectColumnsButton.md

```diff
@@ -65,30 +65,22 @@ const ListActions = () => (
 If you include `<SelectColumnsButton>` in a page that has more than one `<DatagridConfigurable>` (e.g. in a dasboard), you have to link the two components by giving them the same `preferenceKey`:
 
 ```jsx
-const BookList = () => {
-    const { data, total, isLoading } = useGetList('books', {
-        pagination: { page: 1, perPage: 10 },
-        sort,
-    });
-    return (
-        <div>
-            <SelectColumnsButton preferenceKey="postList1" />
-            <DatagridConfigurable
-                preferenceKey="postList1"
-                data={data}
-                total={total}
-                isLoading={isLoading}
-                sort={sort}
-                bulkActionButtons={false}
-            >
-                <TextField source="id" />
-                <TextField source="title" />
-                <TextField source="author" />
-                <TextField source="year" />
-            </DatagridConfigurable>
-        </div>
-    );
-};
+const BookListActions = () => (
+    <TopToolbar>
+        <SelectColumnsButton preferenceKey="postList1" />
+    </TopToolbar>
+);
+
+const BookList = () => (
+    <List actions={<BookListActions />}>
+        <DatagridConfigurable preferenceKey="postList1">
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**issue:** ‏Strange that the  `preferenceKey` is `postList1` than component is called `<BookList>`. Based on the other example (`<PostList>`), we could set `preferenceKey` to `books.datagrid`